### PR TITLE
common/gadi/packages.yaml: add Python as an external

### PR DESCRIPTION
**Pull request by harshula**
* Thanks to Paul Leopardi for finding $NCI_PYTHONX_AS_PYTHON.
* Sort into alphabetical order.

### CODE REVIEW

**Comment by harshula** [comment]
TODO: Need to support Python 2. i.e. `depends_on("python@2.7:")`

File: common/gadi/packages.yaml [hunk 1]
```diff
@@ -290,3 +285,18 @@ packages:
...
+    buildable: false
+  python:
+    externals:
+    - spec: python@3.11.7
+      prefix: /apps/python3/3.11.7
+      modules: [python3/3.11.7]
+      extra_attributes:
+        environment:
+          set:
+            NCI_PYTHONX_AS_PYTHON: python3
```
Comment (aidanheerdegen) [inline common/gadi/packages.yaml:301]: Can you comment in the PR or in the fie why this is necessary?

**Review by aidanheerdegen — APPROVED** [review]
LGTM

**Comment by aidanheerdegen** [comment]
> TODO: Need to support Python 2. i.e. `depends_on("python@2.7:")`

Ugh. Do we though? Is this some old python2 stuff needed for UM suites?

File: common/gadi/packages.yaml (hunk 1 above)
Comment (harshula) [reply common/gadi/packages.yaml:301]: Good idea!

**Comment by harshula** [comment]
Hi @aidanheerdegen , Looks like Spack's Python SPD does *not* support Python2:

```
commit c5883fffd7310f704a8adc2f3666d7539cdfe4e1
Author: Adam J. Stewart <ajstewart426@gmail.com>
Date:   Tue Nov 22 17:02:30 2022 -0600

    Python: drop EOL versions (#33898)
    
    This PR removes [end of life](https://endoflife.date/python) versions of Python from Spack. Specifically, this includes all versions of Python older than 3.7.
    
    See https://github.com/spack/spack/discussions/31824 for rationale. Deprecated in #32615. And #28003.
    
    For anyone using software that relies on Python 2, you have a few options:
    
    * Upgrade the software to support Python 3. The `3to2` tool may get you most of the way there, although more complex libraries may need manual tweaking.
    * Add Python 2 as an [external package](https://spack.readthedocs.io/en/latest/build_settings.html#external-packages). Many Python libraries do not support Python 2, but you may be able to add older versions that did once upon a time.
    * Use Spack 0.19. Spack 0.19 is the last release to officially support Python 3.6 and older
    * Create and maintain your own [custom repository](https://spack.readthedocs.io/en/latest/repositories.html). Basically, you would need a package for Python 2 and any other Python 2-specific libraries you need.
```

**Comment by harshula** [comment]
**Testing**

```
$ spack install python %intel@2021.10.0 target=x86_64
==> python@3.11.7 : has external module in ['python3/3.11.7']
[+] /apps/python3/3.11.7 (external python-3.11.7-lekhnmlnctphwrkq43urxnd4jhq6xoln)

$ spack load python
$ python -V
Python 3.11.9
$ python2 -V
Python 2.7.18
$ python3 -V
Python 3.11.9
$ which python
/bin/python
```

File: common/gadi/packages.yaml [hunk 2]
```diff
@@ -290,3 +285,25 @@ packages:
...
+      # -------------------------------------------------------------------
+      # /apps/Modules/modulefiles/python3-as-python:
+      #
+      # conflict        python2-as-python
+      # setenv          NCI_PYTHONX_AS_PYTHON python3
+      # -------------------------------------------------------------------
+      extra_attributes:
+        environment:
+          set:
+            NCI_PYTHONX_AS_PYTHON: python3
```
Comment (atteggiani) [inline common/gadi/packages.yaml:308]: Maybe you could also add a comment specifying why this env variable needs to be set?

**Review by atteggiani — APPROVED** [review]
LGTM! 
Only added a minor comment.

File: common/gadi/packages.yaml (hunk 2 above)
Comment (harshula) [reply common/gadi/packages.yaml:308]: The comment above it should help:
```
      # $ module show python3-as-python
      # -------------------------------------------------------------------
      # /apps/Modules/modulefiles/python3-as-python:
      #
      # conflict        python2-as-python
      # setenv          NCI_PYTHONX_AS_PYTHON python3
      # -------------------------------------------------------------------
```

**Review by penguian — APPROVED** [review]

**Pull request merged**